### ISSUE TITLE
[+] Collections - Forest doesn't support ActiveStorage

### ISSUE THREAD

**Issue by BenjaminDigeon** [ForestAdmin/forest-rails#253]
When trying to view model data with ActiveStorage in `app.forestadmin.com` the application raise an exception because Forest doesn't know `ActiveStorageAttachmentSerializer`.

<img width="460" alt="capture d ecran 2018-04-09 a 09 25 57" src="https://user-images.githubusercontent.com/685561/38484802-09f2b8b6-3bd8-11e8-8a99-0adfdbf7dad4.png">

Would it be better to not display the collumn insteed of raising a exception ?

Full stacktrace : 

```
[2018-04-09 09:24:11] Forest 🌳🌳🌳  Records Index error: uninitialized constant ForestLiana::UserSpace::ActiveStorageAttachmentSerializer
/Users/ben/.rvm/gems/ruby-2.5.1/gems/bootsnap-1.2.1/lib/bootsnap/load_path_cache/core_ext/active_support.rb:53:in `block in load_missing_constant'
	/Users/ben/.rvm/gems/ruby-2.5.1/gems/bootsnap-1.2.1/lib/bootsnap/load_path_cache/core_ext/active_support.rb:8:in `without_bootsnap_cache'
	/Users/ben/.rvm/gems/ruby-2.5.1/gems/bootsnap-1.2.1/lib/bootsnap/load_path_cache/core_ext/active_support.rb:53:in `rescue in load_missing_constant'
	/Users/ben/.rvm/gems/ruby-2.5.1/gems/bootsnap-1.2.1/lib/bootsnap/load_path_cache/core_ext/active_support.rb:42:in `load_missing_constant'
	/Users/ben/.rvm/gems/ruby-2.5.1/gems/activesupport-5.2.0.rc2/lib/active_support/inflector/methods.rb:285:in `const_get'
	/Users/ben/.rvm/gems/ruby-2.5.1/gems/activesupport-5.2.0.rc2/lib/active_support/inflector/methods.rb:285:in `block in constantize'
	/Users/ben/.rvm/gems/ruby-2.5.1/gems/activesupport-5.2.0.rc2/lib/active_support/inflector/methods.rb:281:in `each'
	/Users/ben/.rvm/gems/ruby-2.5.1/gems/activesupport-5.2.0.rc2/lib/active_support/inflector/methods.rb:281:in `inject'
	/Users/ben/.rvm/gems/ruby-2.5.1/gems/activesupport-5.2.0.rc2/lib/active_support/inflector/methods.rb:281:in `constantize'
	/Users/ben/.rvm/gems/ruby-2.5.1/gems/activesupport-5.2.0.rc2/lib/active_support/core_ext/string/inflections.rb:68:in `constantize'
	/Users/ben/.rvm/gems/ruby-2.5.1/gems/jsonapi-serializers-1.0.0/lib/jsonapi-serializers/serializer.rb:248:in `find_serializer_class'
	/Users/ben/.rvm/gems/ruby-2.5.1/gems/jsonapi-serializers-1.0.0/lib/jsonapi-serializers/serializer.rb:252:in `find_serializer'
	/Users/ben/.rvm/gems/ruby-2.5.1/gems/jsonapi-serializers-1.0.0/lib/jsonapi-serializers/serializer.rb:130:in `block in relationships'
	/Users/ben/.rvm/gems/ruby-2.5.1/gems/jsonapi-serializers-1.0.0/lib/jsonapi-serializers/serializer.rb:109:in `each'
	/Users/ben/.rvm/gems/ruby-2.5.1/gems/jsonapi-serializers-1.0.0/lib/jsonapi-serializers/serializer.rb:109:in `relationships'
	/Users/ben/.rvm/gems/ruby-2.5.1/gems/jsonapi-serializers-1.0.0/lib/jsonapi-serializers/serializer.rb:420:in `serialize_primary'
	/Users/ben/.rvm/gems/ruby-2.5.1/gems/jsonapi-serializers-1.0.0/lib/jsonapi-serializers/serializer.rb:438:in `block in serialize_primary_multi'
	/Users/ben/.rvm/gems/ruby-2.5.1/gems/jsonapi-serializers-1.0.0/lib/jsonapi-serializers/serializer.rb:438:in `map'
	/Users/ben/.rvm/gems/ruby-2.5.1/gems/jsonapi-serializers-1.0.0/lib/jsonapi-serializers/serializer.rb:438:in `serialize_primary_multi'
	/Users/ben/.rvm/gems/ruby-2.5.1/gems/jsonapi-serializers-1.0.0/lib/jsonapi-serializers/serializer.rb:321:in `serialize'
	/Users/ben/.rvm/gems/ruby-2.5.1/gems/forest_liana-2.4.9/app/controllers/forest_liana/application_controller.rb:46:in `serialize_models'
	/Users/ben/.rvm/gems/ruby-2.5.1/gems/forest_liana-2.4.9/app/controllers/forest_liana/resources_controller.rb:134:in `render_jsonapi'
	/Users/ben/.rvm/gems/ruby-2.5.1/gems/forest_liana-2.4.9/app/controllers/forest_liana/resources_controller.rb:22:in `block (2 levels) in index'
	/Users/ben/.rvm/gems/ruby-2.5.1/gems/actionpack-5.2.0.rc2/lib/action_controller/metal/mime_responds.rb:203:in `respond_to'
	/Users/ben/.rvm/gems/ruby-2.5.1/gems/forest_liana-2.4.9/app/controllers/forest_liana/resources_controller.rb:21:in `index'
	/Users/ben/.rvm/gems/ruby-2.5.1/gems/actionpack-5.2.0.rc2/lib/action_controller/metal/basic_implicit_render.rb:6:in `send_action'
	/Users/ben/.rvm/gems/ruby-2.5.1/gems/actionpack-5.2.0.rc2/lib/abstract_controller/base.rb:194:in `process_action'
	/Users/ben/.rvm/gems/ruby-2.5.1/gems/actionpack-5.2.0.rc2/lib/action_controller/metal/rendering.rb:30:in `process_action'
	/Users/ben/.rvm/gems/ruby-2.5.1/gems/actionpack-5.2.0.rc2/lib/abstract_controller/callbacks.rb:42:in `block in process_action'
	/Users/ben/.rvm/gems/ruby-2.5.1/gems/activesupport-5.2.0.rc2/lib/active_support/callbacks.rb:109:in `block in run_callbacks'
	/Users/ben/.rvm/gems/ruby-2.5.1/gems/react-rails-2.4.4/lib/react/rails/controller_lifecycle.rb:31:in `use_react_component_helper'
	/Users/ben/.rvm/gems/ruby-2.5.1/gems/activesupport-5.2.0.rc2/lib/active_support/callbacks.rb:118:in `block in run_callbacks'
	/Users/ben/.rvm/gems/ruby-2.5.1/gems/activesupport-5.2.0.rc2/lib/active_support/callbacks.rb:136:in `run_callbacks'
	/Users/ben/.rvm/gems/ruby-2.5.1/gems/actionpack-5.2.0.rc2/lib/abstract_controller/callbacks.rb:41:in `process_action'
	/Users/ben/.rvm/gems/ruby-2.5.1/gems/actionpack-5.2.0.rc2/lib/action_controller/metal/rescue.rb:22:in `process_action'
	/Users/ben/.rvm/gems/ruby-2.5.1/gems/actionpack-5.2.0.rc2/lib/action_controller/metal/instrumentation.rb:34:in `block in process_action'
	/Users/ben/.rvm/gems/ruby-2.5.1/gems/activesupport-5.2.0.rc2/lib/active_support/notifications.rb:168:in `block in instrument'
	/Users/ben/.rvm/gems/ruby-2.5.1/gems/activesupport-5.2.0.rc2/lib/active_support/notifications/instrumenter.rb:23:in `instrument'
	/Users/ben/.rvm/gems/ruby-2.5.1/gems/activesupport-5.2.0.rc2/lib/active_support/notifications.rb:168:in `instrument'
	/Users/ben/.rvm/gems/ruby-2.5.1/gems/actionpack-5.2.0.rc2/lib/action_controller/metal/instrumentation.rb:32:in `process_action'
	/Users/ben/.rvm/gems/ruby-2.5.1/gems/actionpack-5.2.0.rc2/lib/action_controller/metal/params_wrapper.rb:256:in `process_action'
	/Users/ben/.rvm/gems/ruby-2.5.1/gems/activerecord-5.2.0.rc2/lib/active_record/railties/controller_runtime.rb:24:in `process_action'
	/Users/ben/.rvm/gems/ruby-2.5.1/gems/actionpack-5.2.0.rc2/lib/abstract_controller/base.rb:134:in `process'
	/Users/ben/.rvm/gems/ruby-2.5.1/gems/actionview-5.2.0.rc2/lib/action_view/rendering.rb:32:in `process'
	/Users/ben/.rvm/gems/ruby-2.5.1/gems/actionpack-5.2.0.rc2/lib/action_controller/metal.rb:191:in `dispatch'
	/Users/ben/.rvm/gems/ruby-2.5.1/gems/actionpack-5.2.0.rc2/lib/action_controller/metal.rb:236:in `block in action'
	/Users/ben/.rvm/gems/ruby-2.5.1/gems/forest_liana-2.4.9/app/controllers/forest_liana/router.rb:28:in `call'
	/Users/ben/.rvm/gems/ruby-2.5.1/gems/actionpack-5.2.0.rc2/lib/action_dispatch/routing/mapper.rb:19:in `block in <class:Constraints>'
	/Users/ben/.rvm/gems/ruby-2.5.1/gems/actionpack-5.2.0.rc2/lib/action_dispatch/routing/mapper.rb:48:in `serve'
	/Users/ben/.rvm/gems/ruby-2.5.1/gems/actionpack-5.2.0.rc2/lib/action_dispatch/journey/router.rb:52:in `block in serve'
	/Users/ben/.rvm/gems/ruby-2.5.1/gems/actionpack-5.2.0.rc2/lib/action_dispatch/journey/router.rb:35:in `each'
	/Users/ben/.rvm/gems/ruby-2.5.1/gems/actionpack-5.2.0.rc2/lib/action_dispatch/journey/router.rb:35:in `serve'
	/Users/ben/.rvm/gems/ruby-2.5.1/gems/actionpack-5.2.0.rc2/lib/action_dispatch/routing/route_set.rb:840:in `call'
	/Users/ben/.rvm/gems/ruby-2.5.1/gems/rack-cors-1.0.2/lib/rack/cors.rb:97:in `call'
	/Users/ben/.rvm/gems/ruby-2.5.1/gems/railties-5.2.0.rc2/lib/rails/engine.rb:524:in `call'
	/Users/ben/.rvm/gems/ruby-2.5.1/gems/railties-5.2.0.rc2/lib/rails/railtie.rb:190:in `public_send'
	/Users/ben/.rvm/gems/ruby-2.5.1/gems/railties-5.2.0.rc2/lib/rails/railtie.rb:190:in `method_missing'
	/Users/ben/.rvm/gems/ruby-2.5.1/gems/actionpack-5.2.0.rc2/lib/action_dispatch/routing/mapper.rb:19:in `block in <class:Constraints>'
	/Users/ben/.rvm/gems/ruby-2.5.1/gems/actionpack-5.2.0.rc2/lib/action_dispatch/routing/mapper.rb:48:in `serve'
	/Users/ben/.rvm/gems/ruby-2.5.1/gems/actionpack-5.2.0.rc2/lib/action_dispatch/journey/router.rb:52:in `block in serve'
	/Users/ben/.rvm/gems/ruby-2.5.1/gems/actionpack-5.2.0.rc2/lib/action_dispatch/journey/router.rb:35:in `each'
	/Users/ben/.rvm/gems/ruby-2.5.1/gems/actionpack-5.2.0.rc2/lib/action_dispatch/journey/router.rb:35:in `serve'
	/Users/ben/.rvm/gems/ruby-2.5.1/gems/actionpack-5.2.0.rc2/lib/action_dispatch/routing/route_set.rb:840:in `call'
	/Users/ben/.rvm/gems/ruby-2.5.1/gems/warden-1.2.7/lib/warden/manager.rb:36:in `block in call'
	/Users/ben/.rvm/gems/ruby-2.5.1/gems/warden-1.2.7/lib/warden/manager.rb:35:in `catch'
	/Users/ben/.rvm/gems/ruby-2.5.1/gems/warden-1.2.7/lib/warden/manager.rb:35:in `call'
	/Users/ben/.rvm/gems/ruby-2.5.1/gems/rack-2.0.4/lib/rack/tempfile_reaper.rb:15:in `call'
	/Users/ben/.rvm/gems/ruby-2.5.1/gems/rack-2.0.4/lib/rack/etag.rb:25:in `call'
	/Users/ben/.rvm/gems/ruby-2.5.1/gems/rack-2.0.4/lib/rack/conditional_get.rb:25:in `call'
	/Users/ben/.rvm/gems/ruby-2.5.1/gems/rack-2.0.4/lib/rack/head.rb:12:in `call'
	/Users/ben/.rvm/gems/ruby-2.5.1/gems/actionpack-5.2.0.rc2/lib/action_dispatch/http/content_security_policy.rb:18:in `call'
	/Users/ben/.rvm/gems/ruby-2.5.1/gems/rack-2.0.4/lib/rack/session/abstract/id.rb:232:in `context'
	/Users/ben/.rvm/gems/ruby-2.5.1/gems/rack-2.0.4/lib/rack/session/abstract/id.rb:226:in `call'
	/Users/ben/.rvm/gems/ruby-2.5.1/gems/actionpack-5.2.0.rc2/lib/action_dispatch/middleware/cookies.rb:670:in `call'
	/Users/ben/.rvm/gems/ruby-2.5.1/gems/activerecord-5.2.0.rc2/lib/active_record/migration.rb:559:in `call'
	/Users/ben/.rvm/gems/ruby-2.5.1/gems/actionpack-5.2.0.rc2/lib/action_dispatch/middleware/callbacks.rb:28:in `block in call'
	/Users/ben/.rvm/gems/ruby-2.5.1/gems/activesupport-5.2.0.rc2/lib/active_support/callbacks.rb:98:in `run_callbacks'
	/Users/ben/.rvm/gems/ruby-2.5.1/gems/actionpack-5.2.0.rc2/lib/action_dispatch/middleware/callbacks.rb:26:in `call'
	/Users/ben/.rvm/gems/ruby-2.5.1/gems/actionpack-5.2.0.rc2/lib/action_dispatch/middleware/executor.rb:14:in `call'
	/Users/ben/.rvm/gems/ruby-2.5.1/gems/appsignal-2.5.1/lib/appsignal/rack/rails_instrumentation.rb:17:in `call'
	/Users/ben/.rvm/gems/ruby-2.5.1/gems/better_errors-2.4.0/lib/better_errors/middleware.rb:84:in `protected_app_call'
	/Users/ben/.rvm/gems/ruby-2.5.1/gems/better_errors-2.4.0/lib/better_errors/middleware.rb:79:in `better_errors_call'
	/Users/ben/.rvm/gems/ruby-2.5.1/gems/better_errors-2.4.0/lib/better_errors/middleware.rb:57:in `call'
	/Users/ben/.rvm/gems/ruby-2.5.1/gems/actionpack-5.2.0.rc2/lib/action_dispatch/middleware/debug_exceptions.rb:61:in `call'
	/Users/ben/.rvm/gems/ruby-2.5.1/gems/web-console-3.5.1/lib/web_console/middleware.rb:135:in `call_app'
	/Users/ben/.rvm/gems/ruby-2.5.1/gems/web-console-3.5.1/lib/web_console/middleware.rb:28:in `block in call'
	/Users/ben/.rvm/gems/ruby-2.5.1/gems/web-console-3.5.1/lib/web_console/middleware.rb:18:in `catch'
	/Users/ben/.rvm/gems/ruby-2.5.1/gems/web-console-3.5.1/lib/web_console/middleware.rb:18:in `call'
	/Users/ben/.rvm/gems/ruby-2.5.1/gems/actionpack-5.2.0.rc2/lib/action_dispatch/middleware/show_exceptions.rb:33:in `call'
	/Users/ben/.rvm/gems/ruby-2.5.1/gems/railties-5.2.0.rc2/lib/rails/rack/logger.rb:38:in `call_app'
	/Users/ben/.rvm/gems/ruby-2.5.1/gems/railties-5.2.0.rc2/lib/rails/rack/logger.rb:26:in `block in call'
	/Users/ben/.rvm/gems/ruby-2.5.1/gems/activesupport-5.2.0.rc2/lib/active_support/tagged_logging.rb:71:in `block in tagged'
	/Users/ben/.rvm/gems/ruby-2.5.1/gems/activesupport-5.2.0.rc2/lib/active_support/tagged_logging.rb:28:in `tagged'
	/Users/ben/.rvm/gems/ruby-2.5.1/gems/activesupport-5.2.0.rc2/lib/active_support/tagged_logging.rb:71:in `tagged'
	/Users/ben/.rvm/gems/ruby-2.5.1/gems/railties-5.2.0.rc2/lib/rails/rack/logger.rb:26:in `call'
	/Users/ben/.rvm/gems/ruby-2.5.1/gems/sprockets-rails-3.2.1/lib/sprockets/rails/quiet_assets.rb:13:in `call'
	/Users/ben/.rvm/gems/ruby-2.5.1/gems/actionpack-5.2.0.rc2/lib/action_dispatch/middleware/remote_ip.rb:81:in `call'
	/Users/ben/.rvm/gems/ruby-2.5.1/gems/actionpack-5.2.0.rc2/lib/action_dispatch/middleware/request_id.rb:27:in `call'
	/Users/ben/.rvm/gems/ruby-2.5.1/gems/rack-2.0.4/lib/rack/method_override.rb:22:in `call'
	/Users/ben/.rvm/gems/ruby-2.5.1/gems/rack-2.0.4/lib/rack/runtime.rb:22:in `call'
	/Users/ben/.rvm/gems/ruby-2.5.1/gems/activesupport-5.2.0.rc2/lib/active_support/cache/strategy/local_cache_middleware.rb:29:in `call'
	/Users/ben/.rvm/gems/ruby-2.5.1/gems/actionpack-5.2.0.rc2/lib/action_dispatch/middleware/executor.rb:14:in `call'
	/Users/ben/.rvm/gems/ruby-2.5.1/gems/actionpack-5.2.0.rc2/lib/action_dispatch/middleware/static.rb:127:in `call'
	/Users/ben/.rvm/gems/ruby-2.5.1/gems/rack-2.0.4/lib/rack/sendfile.rb:111:in `call'
	/Users/ben/.rvm/gems/ruby-2.5.1/gems/rack-cors-1.0.2/lib/rack/cors.rb:97:in `call'
	/Users/ben/.rvm/gems/ruby-2.5.1/gems/webpacker-3.4.3/lib/webpacker/dev_server_proxy.rb:18:in `perform_request'
	/Users/ben/.rvm/gems/ruby-2.5.1/gems/rack-proxy-0.6.4/lib/rack/proxy.rb:57:in `call'
	/Users/ben/.rvm/gems/ruby-2.5.1/gems/railties-5.2.0.rc2/lib/rails/engine.rb:524:in `call'
	/Users/ben/.rvm/gems/ruby-2.5.1/gems/puma-3.11.3/lib/puma/configuration.rb:225:in `call'
	/Users/ben/.rvm/gems/ruby-2.5.1/gems/puma-3.11.3/lib/puma/server.rb:624:in `handle_request'
	/Users/ben/.rvm/gems/ruby-2.5.1/gems/puma-3.11.3/lib/puma/server.rb:438:in `process_client'
	/Users/ben/.rvm/gems/ruby-2.5.1/gems/puma-3.11.3/lib/puma/server.rb:302:in `block in run'
	/Users/ben/.rvm/gems/ruby-2.5.1/gems/puma-3.11.3/lib/puma/thread_pool.rb:120:in `block in spawn_thread'
```

**Comment by BenjaminDigeon** [comment]
@arnaudbesnier Hello, any news or feedbacks about this issue ?

**Comment by holden** [comment]
Been wanting to try Forest for a while now, finally found a project, but sadly I think I'd rather forgo Forest (for now) than rely on Refile (or one of the many other less-than-shiny image attachment libraries).

Waiting on pins and needles though.  Thanks. ;-)

**Comment by arnaudbesnier** [comment]
@BenjaminDigeon @holden we are currently investigating this issue ;)

**Comment by ArnaudValensi** [comment]
@BenjaminDigeon @holden This issue seems to be related to the auto "constants" loading of rails. I can reproduce it in development when `config.eager_load` is set to `false` in the file `config/environments/development.rb`. So, setting it to `true` should fix the problem for now. Can you confirm?
Also, by default this value is set to `true` in production, so it shouldn't happen.

**Comment by holden** [comment]
Yeah, that seemed to make it work.  Or at least now the data is loading correctly now.

**Comment by arnaudbesnier** [comment]
@BenjaminDigeon I would like to make it work with `eager_load: false` as it should stay in `development` environment.

Could you bring more details about your issue? Using the default issue template we made could be a good start. We need to at least know which Rails/Forest gem version you use.
And a minimal configuration with a model + has_one_attached/has_many_attached that raise such errors would be great!

@holden, @mathieubrunpicard your own insights would be a plus to help us about this. 😉

**Comment by arnaudbesnier** [comment]
Fixed in this commit: https://github.com/ForestAdmin/forest-rails/commit/70128e52e575a58af6d133838f877e36d7cc5cf5

**Comment by arnaudbesnier** [comment]
Patch available in the new gem version v2.10.3.

Enjoy!

🌲🌲🌲
